native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/CallProcessor.java

```diff
@@ -122,10 +122,9 @@ public static Object nativeCall(BObject client, Object paramSQLString, BArray re
                         PROCEDURE_CALL_RESULT, iteratorObject);
                 Object[] recordDescriptions = recordTypes.getValues();
                 int resultSetCount = 0;
-                if (resultType) {
+                if (resultType && (resultSet = statement.getResultSet()) != null) {
```

Comment:
Can getResultSet check change the pointer to the resultset in case the result set is present? Let's check this logic with other DBs also which returns multiple result sets.

Reply:
In [MySQL](https://github.com/ballerina-platform/module-ballerinax-mysql/blob/4ce4cdfd056a3763c2a4a9af18bf05fa274d057f/ballerina/tests/call-procedures-test.bal#L118
) and [PostgreSQL](https://github.com/ballerina-platform/module-ballerinax-postgresql/blob/041d7fb84dc89927b55daff88eb157f04ab1c1bc/ballerina/tests/call-procedures-test.bal#L62) there are tests for procedure calls which return multiple result sets, both of which pass successfully with the change.

Will add tests for MSSQL as well.